prov/sockets/include/sock.h

```diff
@@ -864,23 +866,12 @@ struct sock_cq {
 	sock_cq_report_fn report_completion;
 };
 
-struct sock_cm_msg_list_entry {
-	uint64_t msg_len;
-	uint8_t retry;
-	uint8_t reserved[7];
-	uint64_t timestamp_ms;
-	struct sockaddr_in addr;
-	struct dlist_entry entry;
-	fid_t fid;
-	struct sock_eq *eq;
-	char msg[0];
-};
-
 struct sock_conn_hdr {
 	uint8_t type;
-	uint8_t reserved[3];
-	int32_t s_port;
-	uint64_t msg_id;
+	uint8_t reserved;
+	uint16_t port;
+	uint32_t cm_data_sz;
```

Comment:
We may be able to use something like __attribute__ ((aligned (8))) after the structure definition.  I'm not sure how well this works across platforms.

Or, maybe this could work? (no idea)

```
    union {
        uint64_t aligned_data[0];
        char cm_data[0];
    };
```c